src/track/globaltrackcache.cpp

```diff
@@ -341,15 +341,13 @@ void GlobalTrackCache::relocateTracks(
                 fileAccess.info(),
                 plainPtr->getId());
         if (!trackRef.hasCanonicalLocation() && trackRef.hasId() && pRelocator) {
-            auto relocatedFileAccess = pRelocator->relocateCachedTrack(
-                    trackRef.getId(),
-                    fileAccess);
-            if (fileAccess.info() != relocatedFileAccess.info()) {
+            auto relocatedFileAccess = pRelocator->relocateCachedTrack(trackRef.getId());
+            if (relocatedFileAccess.info().hasLocation() &&
+                    fileAccess.info() != relocatedFileAccess.info()) {
                 plainPtr->relocate(relocatedFileAccess);
                 trackRef = TrackRef::fromFileInfo(
                         relocatedFileAccess.info(),
                         trackRef.getId());
-                fileAccess = std::move(relocatedFileAccess);
             }
```

Comment:
why the removal? this line looks crucial to me.

Reply:
`fileAccess` has been removed completely from this function.  The new file access is already owned by the track after  `plainPtr->relocate()`. So there is no use for this code line anymore.

Reply:
Ahh sorry, I was only looking at this line in the context of the commit which still contained the fileAccess variable above.